src/openms/include/OpenMS/CHEMISTRY/ISOTOPEDISTRIBUTION/IsoSpecWrapper.h

```diff
@@ -0,0 +1,357 @@
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+// OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+// WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+// OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+// ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// --------------------------------------------------------------------------
+// $Maintainer: Hannes Rost $
+// $Authors: Hannes Rost, Michał Startek, Mateusz Łącki $
+// --------------------------------------------------------------------------
+
+#pragma once
+
+#include <vector>
+
+#include <OpenMS/CONCEPT/Types.h>
+#include <OpenMS/CONCEPT/Constants.h>
+#include <OpenMS/CONCEPT/Exception.h>
+
+#include <OpenMS/KERNEL/Peak1D.h>
+
+#include <OpenMS/CHEMISTRY/EmpiricalFormula.h>
+#include <OpenMS/CHEMISTRY/ISOTOPEDISTRIBUTION/IsotopeDistribution.h>
+
+
+// Override IsoSpec's use of mmap whenever it is available
+#define ISOSPEC_GOT_SYSTEM_MMAN false
+#define ISOSPEC_GOT_MMAN false
+
+#include <OpenMS/../../thirdparty/IsoSpec/IsoSpec/isoSpec++.h>
+
+
+namespace OpenMS
+{
+
+  /**
+    * @brief Interface to the IsoSpec algorithm.
+    * 
+    * Provides an interface to the IsoSpec algorithm.
+    *
+    * @code
+    * Łącki MK, Startek M, Valkenborg D, Gambin A.
+    * IsoSpec: Hyperfast Fine Structure Calculator.
+    * Anal Chem. 2017 Mar 21;89(6):3272-3277. doi: 10.1021/acs.analchem.6b01459.
+    * @endcode
+    *
+    **/
+  class OPENMS_DLLAPI IsoSpecWrapper
+  {
+  /**
+   * @brief Interface for the IsoSpec algorithm - a generator of infinitely-resolved theoretical spectra.
+   */
+
+public:
+
+    /**
+      * @brief Run the algorithm
+      *
+      * This method will run the algorithm with parameters as set up by the constructor. It will return an
+      * IsotopeDistribution containing the observed configurations. The configurations are explicitly stored
+      * in memory, which may become a problem when considering some especially large distributions. If this,
+      * or (a rather small) performance overhead is a concern, then the generator methods (below) should be
+      * used instead.
+      *
+      * This method is provided for convience. As calling that method invalidates the object (the method should
+      * not be called again, nor anythong other than destroying the object should be done with it), the most common
+      * usage pattern of IsoSpecWrapper classes with the run method is:
+      *
+      * IsotopeDistribution dist = IsoSpecWrapperSubclass(...).run();
+      * do something with dist;
+      *
+      * @note Calling this method invalidates the object! In future versions this limitation might be removed.
+      *
+      * @note This method should not be mixed with the generator methods - a given object should only ever have
```

Comment:
ok I think this might be a bit surprising for the user but I guess it is well enough documented.

Reply:
if the state handling of the objects get to complex to properly reset them I guess one could also consider two wrapper for generator and the in memory calculator

Reply:
Resetting is definitely in the pipeline for close future, but not there yet (it's not very complicated per se, but it'll be a pain to test to make sure we're not missing some edge cases). Splitting the generator and the one-shot run() versions into different objects is definitely sensible though, infact it's what we do in our Python bindings, I'l try to do something in this vein this evening. This will also resolve the issue with run() raising exception for Ordered - there just won't be a non-generator version of that.